src/backend/marsha/core/tests/test_api_liveregistration.py

```diff
@@ -338,71 +301,57 @@ def test_api_liveregistration_read_token_user_with_email_none_context_none_r_con
-        # registration with consumer_site
+        # registration has consumer_site
         liveregistration = LiveRegistrationFactory(
-            video=video,
-            lti_user_id="56255f3807599c377bf0e5bf072359fd",
             consumer_site=video.playlist.consumer_site,
+            email="sarah@openfun.fr",
+            lti_user_id="ID1",
+            lti_id="Maths",
+            video=video,
         )
-
-        # token with right context_id and lti_user_id
+        # token has context_id but different one
         jwt_token = AccessToken()
         jwt_token.payload["resource_id"] = str(video.id)
-        jwt_token.payload["context_id"] = str(video.playlist.lti_id)
+        jwt_token.payload["context_id"] = "Maths 02"
+        jwt_token.payload["consumer_site"] = str(video.playlist.consumer_site.id)
+        jwt_token.payload["roles"] = [random.choice(["student", ""])]
+        # token has right email
         jwt_token.payload["user"] = {
-            "email": None,
-            "id": "56255f3807599c377bf0e5bf072359fd",
+            "email": "sarah@openfun.fr",
+            "id": "ID2",
+            "username": "John",
         }
 
         response = self.client.get(
             f"/api/liveregistrations/{liveregistration.id}/",
             HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
         )
 
-        self.assertEqual(response.status_code, 200)
-        self.assertEqual(
-            json.loads(response.content),
-            {
-                "email": liveregistration.email,
-                "id": str(liveregistration.id),
-                "consumer_site": str(video.playlist.consumer_site_id),
-                "lti_user_id": "56255f3807599c377bf0e5bf072359fd",
-                "should_send_reminders": False,
-                "video": str(video.id),
-            },
-        )
+        self.assertEqual(response.status_code, 404)
 
-    def test_api_liveregistration_read_token_user_with_email_none_context_with_ok_userid_wrong(
-        self,
-    ):
-        """Token with no email can't read its registration with wrong lti informations.
+    def test_api_liveregistration_read_token_public_partially_lti(self):
+        """Token with no email and no consumer_site can't read the registration.
 
-        LTI users don't necessary have email. Token users with no email in token can only read
-        registration if the registration has the same duo key lti_user_id and consumer_site.
-        lti_user_id are only unique by consumer_site. Duo lti_user_id/consumer_site must
-        match.
+        This case is not supposed to happen. A LTI connection has necessary a
+        consumer_site, context_id and a lti_user_id defined in the token.
         """
         video = VideoFactory()
+        # registration with a consumer_site
         liveregistration = LiveRegistrationFactory(
             video=video,
-            lti_user_id="56255f3807599c377bf0e5bf0723DIFF",
+            lti_user_id="56255f3807599c377bf0e5bf072359fd",
```

Comment:
Why do you change it ?

Reply:
to show they can read a registration that is not their one